webapp/src/components/link_preview/link_preview.jsx

```diff
@@ -0,0 +1,184 @@
+import PropTypes from 'prop-types';
+import React, {useEffect, useState} from 'react';
+import ReactMarkdown from 'react-markdown';
+import './preview.css';
+
+import Client from 'client';
+import {getLabelFontColor} from '../../utils/styles';
+import {isUrlCanPreview} from 'src/utils/github_utils';
+
+const maxTicketDescriptionLength = 160;
+
+export const LinkPreview = ({embed: {url}, connected}) => {
+    const [data, setData] = useState(null);
+    useEffect(() => {
+        const initData = async () => {
+            if (isUrlCanPreview(url)) {
+                const [owner, repo, type, number] = url.split('github.com/')[1].split('/');
+
+                let res;
+                switch (type) {
+                case 'issues':
+                    res = await Client.getIssue(owner, repo, number);
+                    break;
+                case 'pull':
+                    res = await Client.getPullRequest(owner, repo, number);
+                    break;
+                }
+                if (res) {
+                    res.owner = owner;
+                    res.repo = repo;
+                    res.type = type;
+                }
+                setData(res);
+            }
+        };
+
+        // show is not provided for Mattermost Server < 5.28
+        if (!connected || data) {
+            return;
+        }
+
+        initData();
+    }, [connected, data, url]);
+
+    const getIconElement = () => {
+        const iconProps = {
+            size: 'small',
+            verticalAlign: 'text-bottom',
+        };
+
+        let icon;
+        let colorClass;
+        switch (data.type) {
+        case 'pull':
+            icon = <GitPullRequestIcon {...iconProps}/>;
+
+            colorClass = 'github-preview-icon-open';
+            if (data.state === 'closed') {
+                if (data.merged) {
+                    colorClass = 'github-preview-icon-merged';
+                    icon = <GitMergeIcon {...iconProps}/>;
+                } else {
+                    colorClass = 'github-preview-icon-closed';
+                }
+            }
+
+            break;
+        case 'issues':
+            if (data.state === 'open') {
+                colorClass = 'github-preview-icon-open';
+                icon = <IssueOpenedIcon {...iconProps}/>;
+            } else if (data.state_reason === 'not_planned') {
+                colorClass = 'github-preview-icon-not-planned';
+                icon = <SkipIcon {...iconProps}/>;
+            } else {
+                colorClass = 'github-preview-icon-merged';
+                icon = <IssueClosedIcon {...iconProps}/>;
+            }
+            break;
+        }
+        return (
+            <span className={`pr-2 ${colorClass}`}>
+                {icon}
+            </span>
+        );
+    };
+
+    if (data) {
+        let date = new Date(data.created_at);
+        date = date.toDateString();
```

Comment:
It's best to "return early" to avoid large indents like this

```suggestion
    if (!data) {
        return null;
    }
      
    let date = new Date(data.created_at);
    date = date.toDateString();
```